homework/src/index.js

```diff
@@ -30,18 +32,113 @@
     return elem;
   }
 
+  const root = document.getElementById('root');
+  const header = createAndAppend('header', root, { id: 'header' });
+  const h3 = createAndAppend('h3', header);
+  h3.innerText = 'HYF Repository';
+  const mainDiv = createAndAppend('div', root, { id: 'main' });
+  const div1 = createAndAppend('div', mainDiv, { id: 'div1' });
+  const div2 = createAndAppend('div', mainDiv, { id: 'div2' });
+  const ul = createAndAppend('ul', div1, { id: 'list-container' });
+  const Contributions = createAndAppend('h3', div2)
+  Contributions.innerText = 'Contributions';
+  const ul2 = createAndAppend('ul', div2, { id: 'list-container' });
+
+  function renderRepositoryInfo(info) {
+    const li = createAndAppend('li', ul);
+    const span1 = createAndAppend('span', li, { id: 'title' });
+    span1.innerText = 'Repository: ';
+    const p1 = createAndAppend('p', li);
+    const a = createAndAppend('a', p1, { href: info.RepositoryUrl, target: '_blank' });
+    a.innerText = `${info.Repository}`;
+    const li2 = createAndAppend('li', ul);
+    const span2 = createAndAppend('span', li2, { id: 'title' });
+    span2.innerText = 'Description:  ';
+    const p2 = createAndAppend('p', li2)
+    p2.innerText = info.Description;
+    const li3 = createAndAppend('li', ul);
+    const span3 = createAndAppend('span', li3, { id: 'title' });
+    span3.innerText = 'Forks: ';
+    const p3 = createAndAppend('p', li3)
+    p3.innerText = info.Forks;
+    const li4 = createAndAppend('li', ul);
+    const span4 = createAndAppend('span', li4, { id: 'title' });
+    span4.innerText = 'Updated: ';
+    const p4 = createAndAppend('p', li4)
+    p4.innerText = info.Updated;
+  }
+
+  function renderContributorsInfo(info) {
+    fetchJSON(info.ContributorsUrl)
+      .then(data => {
+        ul2.innerText = '';
+        data.forEach(elem => {
+          const li = createAndAppend('li', ul2);
+          createAndAppend('img', li, { src: elem.avatar_url });
+          const a = createAndAppend('a', li, { href: elem.html_url, target: '_blank' });
+          a.innerText = `${elem.login}`;
+          const span = createAndAppend('span', li);
+          span.innerText = `${elem.contributions}`;
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', div2, { text: err.message, class: 'alert-error' });
+      });
+  }
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+    fetchJSON(url)
+      .then(data => {
```

Comment:
`data` -> `repositories` !

You could sort the array of the repositories at this point. There is no need to create a separate array (the one you called `queries`,  you could use the `repositories` array instead. Your `queries` array is just a subset of fields of the `repositories` array. Nothing is gained from creating this extra array, it is just adding 'noise'.